torch_geometric/nn/to_hetero_transformer.py

```diff
@@ -286,27 +289,47 @@ def call_global_pooling_module(self, node: Node, target: Any, name: str):
     def call_module(self, node: Node, target: Any, name: str):
         if self.is_graph_level(node):
             return
-
-        # Add calls to node type-wise or edge type-wise modules.
         self.graph.inserting_after(node)
-        for key in self.metadata[int(self.is_edge_level(node))]:
-            args, kwargs = self.map_args_kwargs(node, key)
-            out = self.graph.create_node('call_module',
-                                         target=f'{target}.{key2str(key)}',
-                                         args=args, kwargs=kwargs,
-                                         name=f'{name}__{key2str(key)}')
-            self.graph.inserting_after(out)
+        if hasattr(self.module, name):
+            submod = getattr(self.module, name)
+            is_heterolin = is_linear(submod)
+        else:
+            split_name = name.split('_')
+            submod = getattr(self.module, '_'.join(split_name[:-1]))
+            # handle iterable Modules (dict/list/sequential) containing Linear
+            if is_iterable_module(submod):
+                try:
+                    selected_subsubmod = submod[split_name[-1]]
+                except TypeError:
+                    selected_subsubmod = submod[int(split_name[-1])]
+                is_heterolin = is_linear(selected_subsubmod)
+            else:
+                is_heterolin = False
+        if is_heterolin:
+            self.add_heterolin_to_graph(node, target, name)
+        else:
+            # Add calls to node type-wise or edge type-wise modules.
+            self.add_nonlin_to_graph(node, target, name)
 
     def call_method(self, node: Node, target: Any, name: str):
+        # Add calls to node type-wise or edge type-wise methods.
         if self.is_graph_level(node):
             return
-
-        # Add calls to node type-wise or edge type-wise methods.
         self.graph.inserting_after(node)
+        if torch_geometric.typing.WITH_PYG_LIB:
+            # Addresses:
+            # "RuntimeError: Output 0 of SplitWithSizesBackward0
+            # is a view and is being modified inplace."
+            # Cause:
+            # split_with_sizes for pyg-lib.ops.segment_matmul returns a view.
+            # Using relu_ or other in place on this triggers the RuntimeError.
```

Comment:
Then let's just not use `relu_`.

Reply:
yes thats what my code does. if there is a relu_(or any in place will have the same issue) in the original homogeneous model then I replace it with the equivalent out of place in the final. this is needed to solve that RuntimeError. had mentioned in slack which is at the top of the comment. In general I believe my PR should accelerate to_hetero with zero code changes from the user needed or any changes in user experience and object behavior. I have put in a great deal of effort to make sure this is the case.